### ISSUE TITLE
sam9x60 : qspi write error

### ISSUE THREAD

**Issue by FabienLehoussel** [linux4sam/u-boot-at91#10]
On 2019.04 branch, qspi write doesn't work

I can’t save my env. Write function return ok, but nothing seems to be writen on NOR.

I had CONFIG_CMD_SF_TEST to uboot to test spi flash : 

```console
U-Boot 2019.04-linux4sam_6.2-icp-rc1-dirty (Jan 13 2020 - 09:32:53 +0100)

CPU: SAM9X60
Crystal frequency:       24 MHz
CPU clock        :      600 MHz
Master clock     :      200 MHz
DRAM:  256 MiB
NAND:  512 MiB
MMC:
Loading Environment from SPI Flash... SF: Detected sst26vf064b with page size 256 Bytes, erase size 4 KiB, total 8 MiB
*** Warning - bad CRC, using default environment

In:    serial
Out:   serial
Err:   serial
Net:   eth0: ethernet@f802c000
Hit any key to stop autoboot:  0

U-Boot> env save
Saving Environment to SPI Flash... Erasing SPI flash...Writing to SPI flash...done
OK
U-Boot>

U-Boot> sf probe
SF: Detected sst26vf064b with page size 256 Bytes, erase size 4 KiB, total 8 MiB

U-Boot> sf test 0x140000 0x1000
SPI flash test:
0 erase: 17 ticks, 235 KiB/s 1.880 Mbps
1 check: 0 ticks, 4096000 KiB/s 32768.000 Mbps
2 write: 17 ticks, 235 KiB/s 1.880 Mbps
3 read: 1 ticks, 4000 KiB/s 32.000 Mbps
Verify failed at 0, good data:
00000000: b8 00 00 ea 14 f0 9f e5 14 f0 9f e5 14 f0 9f e5    ................
00000010: 14 f0 9f e5 1b 30 0a 00 14 f0 9f e5 14 f0 9f e5    .....0..........
00000020: 60 00 f0 23 c0 00 f0 23 20 01 f0 23 80 01 f0 23    `..#...# ..#...#
00000030: e0 01 f0 23 40 02 f0 23 a0 02 f0 23 ef be ad de    ...#@..#...#....
Bad data:
00000000: ff ff ff ff ff ff ff ff ff ff ff ff ff ff ff ff    ................
00000010: ff ff ff ff ff ff ff ff ff ff ff ff ff ff ff ff    ................
00000020: ff ff ff ff ff ff ff ff ff ff ff ff ff ff ff ff    ................
00000030: ff ff ff ff ff ff ff ff ff ff ff ff ff ff ff ff    ................
Test failed

U-Boot>
```

**Comment by ehristev** [comment]
Can you please retest on branch u-boot-2020.01-at91
Fixes have been pushed there for your issue.
Thanks for your patience !

**Comment by ehristev** [comment]
Fixes also available in 2019.04-at91 branch
